### PR TITLE
Update Reactor docs with info on execution scope

### PR DESCRIPTION
**Pull request by whiteinge**

### CODE REVIEW

File: doc/topics/reactor/index.rst [hunk 1]
```diff
@@ -98,6 +97,90 @@ can be called:
...
 
+The Goal of Writing Reactor SLS Files
+=====================================
+
+Reactor SLS files share the familiar syntax from Salt States but there are
+important differences. The goal of a Reactor file is to process a Salt event as
+quickly as possible and then to optionally start a **new** process in response.
+
+They must be quick because Reactor sls files are matched and rendered
+sequentially in a single process. Long-running Reactor files will cause other
```
Comment (cachedout) [inline doc/topics/reactor/index.rst]: This is sort of misleading. The _rendering_ must be quick, but the execution doesn't have to be. (You do point that out down below, but this threw me off.)

File: doc/topics/reactor/index.rst [hunk 2]
```diff
@@ -98,6 +97,90 @@ can be called:
...
+reactions to queue behind the current one. The process that a Reactor file
+starts is handed off to a pool of worker threads. That process can be a long
+running, complex operation.
+
+Maybe that is a process that sends a command down to minions which is a quick
+fire-and-forget, or maybe it's complex Orchestrate run that sends commands to
+some minions, waits for the response, then later sends commands to other
+minions.
+
+tl;dr: Reactor SLS files MUST be simple and quick. The new job that they start
```
Comment (cachedout) [inline doc/topics/reactor/index.rst]: Same point here as above.

**Comment by cachedout** [comment]
@whiteinge Suggested a few tweaks, if you wouldn't mind having a look.

**Comment by whiteinge** [comment]
Good feedback. That paragraph was too unfocused.

**Pull request merged**